packages/date-picker/src/vaadin-date-picker-overlay-content-mixin.js

```diff
@@ -1009,18 +1020,18 @@ export const DatePickerOverlayContentMixin = (superClass) =>
     }
 
     /** @private */
-    _dateAllowed(date, min = this.minDate, max = this.maxDate) {
-      return (!min || date >= min) && (!max || date <= max);
+    _dateAllowed(date, min = this.minDate, max = this.maxDate, isDateDisabled = this.isDateDisabled) {
+      return (!min || date >= min) && (!max || date <= max) && (!isDateDisabled || !isDateDisabled(date));
```

Comment:
This should use the same `typeof` check as above.
```suggestion
      const dateIsDisabled = typeof isDateDisabled === 'function' ? isDateDisabled(date) : false;
      return (!min || date >= min) && (!max || date <= max) && (!dateIsDisabled);
```

Reply:
updated